test/checkoutput/README.md

```diff
@@ -0,0 +1,45 @@
+# checkoutput tool
+
+The checkoutput tool parses a text or markdown that has a command and the expected output sequence, runs the command and checks the output of the command against the expected output. 
+
+## Usage
+```
+<tool> <path/to/directory> -v X
+
+If directory path is not specified then the current working directory is taken as the root.
+-v specifies the klog logging level for verbosity. Optional parameter.
+```
+
+The tool has 2 major functions
+
+1. Identify all directories with the main packages and corresponding markdown files that has the Run/Verify blocks.
+2. For each such directory that is identified:
+2.1. Parse and capture the Run/Verify blocks
+2.2. Run and verify the output of the blocks captured in the previous step
+
+## Example
+
+The typical use case is a directory with the package main and a markdown file that has the following syntax. In the below example the go will run the main package and the output of it will be checked against the execpted output. The current working directory will be set to the directory containing the markdown file.
+
+Run:
+```console
+go run .
+```
+Expected output:
+```
+I0605 22:03:07.224293 3228948 logger.go:58] Log using Infof, key: value
```

Comment:
This example demonstrates a key problem: the time stamps will never match.

Reply:
We have to support plain text matching and regular expression matching.

Reply:
Yes, I noticed that too when I ran it. I will add regular expression matching.